### PR TITLE
Fixed BUILDIFIER_PATH and BUILDOZER_PATH env var expansion in check_format.py and envoy_build_fixer.py scripts

### PR DESCRIPTION
**Pull request by dmitri-d**
Signed-off-by: Dmitri Dolguikh <ddolguik@redhat.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: Fixes BUILDIFIER_PATH and BUILDOZER_PATH env var expansion when scripts have to fall back on $GOLANG path. Python's os.getenv returns string in the "default value" parameter verbatim, which results in scipts not finding buildozer and buildifier binaries.
Risk Level: low
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: tools/check_format.py [hunk 1]
```diff
@@ -71,8 +71,8 @@
 GRPC_INIT_WHITELIST = ("./source/common/grpc/google_grpc_context.cc")
 
 CLANG_FORMAT_PATH = os.getenv("CLANG_FORMAT", "clang-format-9")
-BUILDIFIER_PATH = os.getenv("BUILDIFIER_BIN", "$GOPATH/bin/buildifier")
-BUILDOZER_PATH = os.getenv("BUILDOZER_BIN", "$GOPATH/bin/buildozer")
+BUILDIFIER_PATH = (os.getenv("BUILDOZER_BIN") or (os.path.expandvars("$GOPATH/bin/buildifier") if os.getenv("GOPATH") else "Can't find buildifier binary"))
```
Comment (lizan) [inline tools/check_format.py]: can we just fallback to `shutil.which("buildifier")`? So if it exists in PATH we don't fail it.

Comment (htuch) [reply tools/check_format.py]: That sounds like a good idea.

**Comment by dmitri-d** [comment]
* use shutil.which as a fallback

File: tools/check_format.py [hunk 2]
```diff
@@ -71,8 +72,8 @@
 GRPC_INIT_WHITELIST = ("./source/common/grpc/google_grpc_context.cc")
 
 CLANG_FORMAT_PATH = os.getenv("CLANG_FORMAT", "clang-format-9")
-BUILDIFIER_PATH = os.getenv("BUILDIFIER_BIN", "$GOPATH/bin/buildifier")
-BUILDOZER_PATH = os.getenv("BUILDOZER_BIN", "$GOPATH/bin/buildozer")
+BUILDIFIER_PATH = (os.getenv("BUILDOZER_BIN") or (os.path.expandvars("$GOPATH/bin/buildifier") if os.getenv("GOPATH") else shutil.which("buildifier")))
```
Comment (htuch) [inline tools/check_format.py]: LGTM, although you don't need the outer parens surrounding the expressions. Thanks.

**Comment by dmitri-d** [comment]
* removed extraneous parens

**Comment by mattklein123** [comment]
Please merge master and fix format. Thank you!

/wait

**Comment by dmitri-d** [comment]
* rebased

Formatting errors are due to numerous buildozer failures: "flag provided but not defined: -mode".

File: tools/check_format.py [hunk 3]
```diff
@@ -74,8 +75,8 @@
 GRPC_INIT_WHITELIST = ("./source/common/grpc/google_grpc_context.cc")
 
 CLANG_FORMAT_PATH = os.getenv("CLANG_FORMAT", "clang-format-9")
-BUILDIFIER_PATH = os.getenv("BUILDIFIER_BIN", "$GOPATH/bin/buildifier")
-BUILDOZER_PATH = os.getenv("BUILDOZER_BIN", "$GOPATH/bin/buildozer")
+BUILDIFIER_PATH = os.getenv("BUILDOZER_BIN") or (os.path.expandvars("$GOPATH/bin/buildifier") if os.getenv("GOPATH") else shutil.which("buildifier"))
```
Comment (htuch) [inline tools/check_format.py]: While this hasn't merged yet, I recently identified a need to access this path elsewhere; can we move this to a separate module that we can import?

**Comment by dmitri-d** [comment]
* rebased
* moved path getters into a dedicated module

@htuch: ping

File: tools/check_format.py [hunk 4]
```diff
@@ -12,6 +12,8 @@
 import stat
 import sys
 import traceback
+import shutil
```
Comment (htuch) [inline tools/check_format.py]: Nit: can you remove shutil from this and `envoy_build_fixer.py`? Good to go otherwise, thanks!
/wait

Comment (dmitri-d) [reply tools/check_format.py]: done

**Comment by htuch** [comment]
@dmitri-d format check failed.

**Comment by dmitri-d** [comment]
ping @htuch. The failures are in stats:thread_local_store_test and fuzz tests, don't appear to be related to the changes in the PR.

**Comment by lizan** [comment]
retriggered test, thanks!

**Pull request merged**